db-engine/src/main/java/com/splicemachine/db/catalog/types/RoutineAliasInfo.java

```diff
@@ -356,12 +379,14 @@ public void writeExternal( ObjectOutput out )
 		out.writeShort(sqlOptions);
 		out.writeObject(returnType);
 		out.writeBoolean(calledOnNullInput);
-		out.writeInt(0); // future expansion
+		out.writeInt(1); // future expansion
 		if (parameterCount != 0) {
 			ArrayUtil.writeArrayItems(out, parameterNames);
 			ArrayUtil.writeArrayItems(out, parameterTypes);
 			ArrayUtil.writeIntArray(out, parameterModes);
 		}
+		out.writeObject(language);
+		out.writeObject(compiledPyCode);
```

Comment:
Prefer to put these two lines inside the "if (expansionNum ==1)" statement.

Reply:
If the user is using the version with Python stored procedure, he/she always needs to write these two fields even though they are null. However, if the user is not using this version, then these two fields will not be written as the code will not be here.

Therefore, I think it is OK to leave the code as it is. May I know what is your concern?